### ISSUE TITLE
Numeric object keys

### ISSUE THREAD

**Issue by K4rakara** [gura-conf/gura#15]
`serde_gura` will serialize the following without question, but it will throw an error if you attempt to deserialize it:

```gura
resumable_shards:
    0:
        session_id: "xxxx"
        sequence: 2
```

It panics [here](https://github.com/gura-conf/serde-gura/blob/00c64748bd216e544154a55327dd927fa1cd7db3/src/error.rs#L67).

**Comment by Genarito** [comment]
Hi @K4rakara!

Thanks for your report! I'm going to fix it during the next week :)

**Comment by Genarito** [comment]
Excuse me @K4rakara , could you provide the code you're trying, please? So I can replicate exactly

**Comment by Genarito** [comment]
Hi! I've been trying but I couldn't replicate the error. The following code serializes/deserializes correctly:

```rust
use std::collections::HashMap;
use serde_derive::{Deserialize, Serialize};
use serde_gura::Result;

#[derive(Debug, Deserialize, Serialize, PartialEq)]
struct Values {
    session_id: String,
	sequence: u32
}

#[derive(Debug, Deserialize, Serialize, PartialEq)]
struct Shards {
    resumable_shards: HashMap<String, Values>,
}


fn main() -> Result<()> {
    let gura_string = r#"resumable_shards:
    0:
        session_id: "xxxx"
        sequence: 2"#;

    let shards: Shards = serde_gura::from_str(gura_string)?;
	println!("{:?}", shards);
	println!("{}", serde_gura::to_string(&shards).unwrap());

    Ok(())
}
```

Anyway, I would like to know how the structure you are deserializing is, since [the error line](https://github.com/gura-conf/serde-gura/blob/00c64748bd216e544154a55327dd927fa1cd7db3/src/error.rs#L67) indicates that there is an `Error` variant that I didn't consider in the implementation of the `Display` trait for `serde_gura::Error`.

**Comment by K4rakara** [comment]
I'll see if I can repo, though I've changed my code a fair deal since then, so I'll have to test it in an isolated example.

**Comment by Genarito** [comment]
I am pleased to announce that I have now released version `0.1.4` of the [crate](https://crates.io/crates/serde_gura) :partying_face: I noticed that it was throwing an error when the `HashMap` key was numeric instead of String. With the new version of Serde Gura you can now use the example above but with `usize`, `isize`, `u8`, `u32`, `i32`, `i32`, `u64`, `i64`, etc as the key type!

```rust
// ...
#[derive(Debug, Deserialize, Serialize, PartialEq)]
struct Shards {
    resumable_shards: HashMap<usize, Values>, // NEW: numeric key!
}
// ...
```

Thanks for reporting and using Gura! I will close the issue, feel free to reopen it if you consider that the problem is still not solved.

Regards! :smile:
